reascripts/ReaSpeech/source/TranscriptExporter.lua

```diff
@@ -34,23 +34,108 @@ function TranscriptExporter:init()
+    input_width = self.FILE_WIDTH,
+    on_set = function() self:update_target_filename_ui() end,
   })
 
+  -- invisible state to manage the UX around the target filename
+  self.is_full_path = Storage.memory(false)
+  self.target_filename_exists = Storage.memory(false)
+  self.has_extension = Storage.memory(false)
+
+  -- this is the value that will actually be used after the
+  -- file_selector contents are processed (relative vs full path,
+  -- auto extension, etc.)
+  self.target_filename = Storage.memory('')
+
+  self.apply_extension = ReaSpeechCheckbox.new {
+    default = true,
+    label_long = 'Apply Extension',
+    disabled_if = function() return self.has_extension:get() end,
+    changed_handler = function()
+      if not self.apply_extension then return end
+      self:update_target_filename_ui()
+    end
+  }
+
   self.alert_popup = AlertPopup.new {}
 end
 
+function TranscriptExporter:open()
+  self:reset_form()
+end
+
+function TranscriptExporter:reset_form()
+  self.file_selector:set('')
+  self.apply_extension:set(true)
+  self:update_target_filename_ui()
+end
+
+function TranscriptExporter:clear_target_filename()
+  self.target_filename:set('')
+  self.target_filename_exists:set(false)
+  self.has_extension:set(false)
+  self.is_full_path:set(false)
+end
+
+function TranscriptExporter:update_target_filename_ui()
+  local specified_path = self.file_selector:value()
+
+  if specified_path == '' then
+    self:clear_target_filename()
+    return
+  end
+
+  local full_path = PathUtil.get_real_path(specified_path)
+
+  self.has_extension:set(PathUtil.has_extension(full_path))
+  self.is_full_path:set(PathUtil.is_full_path(full_path))
+
+  -- automatically turn off the auto-extension if user enters
+  -- in a filename with an extension
+  if self.has_extension:get() and self.apply_extension:value() then
+    self.apply_extension:set(false)
+  end
+
+  -- automatically apply default extension for format, if desired
+  if self.apply_extension:value() then
+    local extension = self.export_formats:selected_format().extension
+    full_path = PathUtil.apply_extension(full_path, extension)
+  end
+
+  self.target_filename:set(full_path)
+
+  self.target_filename_exists:set(reaper.file_exists(self.target_filename:get()))
+end
+
 function TranscriptExporter:show_success()
   self.alert_popup.onclose = function ()
     self.alert_popup.onclose = nil
     self:close()
   end
-  self.alert_popup:show('Export Successful', 'Exported ' .. self.export_formats:selected_key() .. ' to: ' .. self.file_selector:value())
+
+  self.alert_popup:show('Export Successful', function()
+    local file_path = self.target_filename:get()
+    local filename = PathUtil.get_filename(self.target_filename:get())
+
+    ImGui.Text(ctx, 'Exported ' .. self.export_formats:selected_key() .. ' to: ')
+    ImGui.SameLine(ctx)
+
+    Widgets.link(filename, function()
+      ExecProcess.via_tempfile(PathUtil.get_reveal_command(file_path)):no_wait()
```

Comment:
Do we need via_tempfile for this? I thought using /usr/bin/env would get around that.

Reply:
good question, i'll double-check

Reply:
worked for me without it, thanks for calling that out 🚀